transcom/mymove
# I 12278 allowable weight in weight ticket is reverted to default value (net weight)  if it is the first field selected, then de focused

Summary
Note: I 12278 addresses stories B-18163 & B-18300, this covers a few bugs that were found after the 18163 PR was merged.

The fixes include:

- Selecting between empty and full weight causes validation to break and no longer respond to user input regardless if the entered value is valid or not.
- Selecting off of empty weight to full weight resets the empty weight field to empty.
- Deselecting any of the weight fields with an invalid value doesn't persist the error message (error shows, but dissapears when fields are no longer focused).
- Allowable weight shows 'NaN' in validation error message after clicking off of it about 3 times, regardless of actual value.

Testing will be the same as the 18163 story with minor differences.

##How to test

Login as a service counselor (under officelocal)
Select PPM Closeout find a move that has a PPM shipment that is over weight (above the allowable weight entitled)
Select 'Review Shipment Weights'
Click Review Documents under Weight moved by customer.
5a. The Allowable Weight should reflect the equivalent of NetWeight by default (if there is no allowable weight already set for this weight ticket. Adjust this to 0, to display a validation error, then clear the field completely. Both cases should show an error. Then adjust back to the NetWeight default.
5b.Adjust the Full weight so that Full weight - Empty weight is <= allowable weight under PPM Net Weight. You can also do this by reducing full weight by "to reduce excess weight" under Net Weight. Note: The net weight value is updated after losing focus of either full or empty weight, but not allowable weight. The form is responsive immediately to field validation and should reflect an invalid value without losing focus.
5b. There should be no direct way to adjust the net weight altering the values within full or empty weight should be the only way to modify the net weight. 
5c. Adjust the full weight to 0, then to empty weight -1, then equal to empty weight, then empty weight + 1.  The first 3. cases should show a validation error, but the last should not.
when altering the weight for full or empty weight, the net weight should adjust (after the field is un-focused).
after reducing the weight, the overweight notification should disappear.
after clicking confirm and continue navigate to Move details, the "This move has excess weight" banner should not be visible.
5d. Click off of empty, full, and allowable weight 3 times each. Then click between empty and full weight to edit the fields 4 times each.
5e. While there is a validation error, the form should not be able to be submitted.
